src/providers/AuthCheckProvider.tsx

```diff
@@ -14,6 +16,14 @@ export const AuthCheckProvider: React.FC<AuthCheckProviderProps> = ({
   children,
 }) => {
   const { panHandlers } = useAuthCheck();
+  const { account } = useAuthenticationStore();
+
+  useEffect(() => {
+    // Checking for public key to make sure the account is loaded
+    if (account?.publicKey) {
+      analytics.identifyUser();
```

Comment:
is it important to wait for the publicKey here since the analytics userId is not tied to it? is the extension also waiting for the publicKey to identify users?

Reply:
I see we are still using `account` as dep here. Is that really necessary?
I'm thinking about a few downsides:
- I'm assuming it'll only start sending events after user has logged in, which means we'd miss events like the recovery phrase creation and verification. Is that the case?
- It'll call `identifyUser()` again every time the user switches between its wallet accounts since it'll modify the `account` dependency on this `useEffect`
- Is `AuthCheckProvider` the best place to call `identifyUser()` considering it shouldn't rely on auth statuses? Would `RootNavigator` be a good place?